onmt/translate/translation.py

```diff
@@ -62,21 +61,14 @@ def from_batch(self, translation_batch):
                     key=lambda x: x[-1])))
 
         # Sorting
-        inds, perm = torch.sort(batch.indices.data)
-        data_type = self.data.data_type
-        if data_type == 'text':
-            src = batch.src[0].data.index_select(1, perm)
-        else:
-            src = None
-
-        if self.has_tgt:
-            tgt = batch.tgt.data.index_select(1, perm)
-        else:
-            tgt = None
+        inds, perm = torch.sort(batch.indices)
+        dtype = self.data.data_type
+        src = batch.src[0].index_select(1, perm) if dtype == 'text' else None
+        tgt = batch.tgt.index_select(1, perm) if self.has_tgt else None
 
         translations = []
         for b in range(batch_size):
-            if data_type == 'text':
+            if dtype == 'text':
                 src_vocab = self.data.src_vocabs[inds[b]] \
```

Comment:
I don't like too much dtype since it's a Pytorch word.

Reply:
This is a good point. Fixed.